dali/kernels/test/scatter_gather_test.cc

```diff
@@ -88,79 +88,86 @@ class ScatterGatherTest : public testing::Test {
       CUDA_CALL(cudaMemset(dst, c, size));
     }
   }
-};
 
-TYPED_TEST_SUITE_P(ScatterGatherTest);
+  void CopyTestImpl(size_t total_size, size_t max_l, size_t max_block) {
+    std::vector<char> in(total_size);
+    std::vector<char> out(total_size);
+    unsigned seed = 42;
 
-TYPED_TEST_P(ScatterGatherTest, Copy) {
-  const size_t max_l = 1024;
-  std::vector<char> in(1<<20);
-  std::vector<char> out(1<<20);
-  unsigned seed = 42;
+    using kind = std::conditional_t<std::is_same<T, ScatterGatherCPU>::value,
+                                    mm::memory_kind::host, mm::memory_kind::device>;
 
-  using kind = std::conditional_t<std::is_same<TypeParam, ScatterGatherCPU>::value,
-                                  mm::memory_kind::host, mm::memory_kind::device>;
+    auto in_ptr = mm::alloc_raw_unique<char, kind>(in.size());
+    auto out_ptr = mm::alloc_raw_unique<char, kind>(out.size());
 
-  auto in_ptr = mm::alloc_raw_unique<char, kind>(in.size());
-  auto out_ptr = mm::alloc_raw_unique<char, kind>(out.size());
+    std::vector<detail::CopyRange> ranges;
+    std::vector<detail::CopyRange> back_ranges;
 
-  std::vector<detail::CopyRange> ranges;
-  std::vector<detail::CopyRange> back_ranges;
+    size_t i = 0, j = 0;
+    for (;;) {
```

Comment:
```suggestion
    while (true) {
```

Reply:
This could start a very long and useless discussion ;)